File: app/schemas/clinic_admin.py
"""Clinic admin schemas."""

from __future__ import annotations

from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class DoctorUserDTO(BaseModel):
    id: int
    email: str
    full_name: str | None
    role: str
    is_active: bool
    created_at: datetime | None = None


class DoctorListResponse(BaseModel):
    items: list[DoctorUserDTO]


class DoctorUpdateRequest(BaseModel):
    role: str | None = None
    is_active: bool | None = None


class AuditLogItemDTO(BaseModel):
    id: int
    clinic_id: int
    created_at: datetime | None
    actor_id: int | None
    actor_name: str | None
    actor_role: str | None
    action: str
    entity: str
    entity_id: str | None
    diff_json: dict[str, Any] | None
    request_id: str | None


class AuditLogListResponse(BaseModel):
    items: list[AuditLogItemDTO]
    limit: int
    offset: int
    total: int


class ClinicDashboardKpis(BaseModel):
    total_claims: int
    draft_claims: int
    finalized_claims: int
    active_doctors: int


class ClinicDashboardInsurer(BaseModel):
    insurance_company_id: int
    name: str
    claim_count: int


class ClinicDashboardTimeseries(BaseModel):
    date: date
    count: int


class ClinicDashboardRange(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    from_: date = Field(..., alias="from")
    to: date


class ClinicDashboardResponse(BaseModel):
    range: ClinicDashboardRange
    kpis: ClinicDashboardKpis
    top_insurers: list[ClinicDashboardInsurer]
    claims_timeseries: list[ClinicDashboardTimeseries]
    ai_timeseries: list[ClinicDashboardTimeseries]
    recent_activity: list[AuditLogItemDTO]

File: frontend-dev/src/types/clinicAdmin.ts
export type DoctorUserDTO = {
  id: number;
  email: string;
  full_name?: string | null;
  role: "doctor" | "chief_doctor" | "clinic_admin" | "platform_staff_admin";
  is_active: boolean;
  created_at?: string | null;
};

export type AuditLogItemDTO = {
  id: number;
  created_at?: string | null;
  actor_id?: number | null;
  actor_name?: string | null;
  actor_role?: string | null;
  action: string;
  entity: string;
  entity_id?: string | null;
  diff_json?: Record<string, unknown> | null;
  request_id?: string | null;
};

export type ClinicDashboardDTO = {
  range: { from: string; to: string };
  kpis: {
    total_claims: number;
    draft_claims: number;
    finalized_claims: number;
    active_doctors: number;
  };
  top_insurers: Array<{ insurance_company_id: number; name: string; claim_count: number }>;
  claims_timeseries: Array<{ date: string; count: number }>;
  ai_timeseries: Array<{ date: string; count: number }>;
  recent_activity: AuditLogItemDTO[];
};

export type ClinicAuditLogResponseDTO = {
  items: AuditLogItemDTO[];
  limit: number;
  offset: number;
  total: number;
};

export type ClinicDoctorsResponseDTO = {
  items: DoctorUserDTO[];
};
